### PR TITLE
Add city details and top cities list

### LINKED ISSUE #274: City Exploration of Hotspots
A dedicated city page on Explorer could be a great way to see how coverage is spread across the world (it could also help track growth in different cities).

Thinking of these as the main features:
- A cities page sorted by those with the most Hotspots, showing the number of Hotspots next to the city (and possibly 30 day growth stats for Hotspots in the city)
- The link for the cities could be, for example, "/cities/chicago," or "/cities/portland" and feature a minimap with statistics of total Hotspots, growth for the city, and potentially other stats like Data Credits used in the city, total $HNT rewards for Hotspots in the city over 30 days, or average $HNT rewards per Hotspot in the city over the past day
- Each city could have a "minimap" like on the Explorer homepage, and then when you click on the minimap, it takes you to "/coverage/cities/chicago" or "/coverage/cities/portland" to look at the coverage map, starting from a centrally focused view of the city

On top of the current spread of Hotspots and the stats shown for specific areas, I also think this would also help with time-lapses and visual growth of the network in different parts of the world with these city-focused maps and rewards planning for those hoping to place a Hotspot in a specific city and wondering what current rewards are like.

### LINKED ISSUE #673: [Bug]: Can't search for a city
### What happened?

When searching for a city, suggestions pop up while / after typing.  Choosing any one of them that is labeled as a city results in the following error screen:

Application error: a client-side exception has occurred (developer guidance).

image

### URL

https://explorer.helium.com/

### What browsers are you seeing the problem on?

Chrome, Microsoft Edge

### Do you see the issue on Mobile or Desktop?

Desktop

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by danielcolinjames**
building on #620 

also fixes #673 

adds an info box when you visit a city URL:
Screen Shot 2021-08-24 at 7 28 39 PM

and also an infinite-scrolling top cities list:
Screen Shot 2021-08-24 at 11 44 04 PM

https://helium-explorer-pr-633.herokuapp.com/hotspots/cities

### CODE REVIEW

**Comment by ItsCEED** [comment]
Maybe add SORT option in Hotspots tab so we can sort by most earnings, elevation, scale?

File: components/InfoBox/CityDetailsInfoBox.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+      console.log(city)
+      
+      const geometry = await fetchApi('v1')(
+        '/cities/search?' +
+        qs.stringify({
+          term: [city.longCity, city.longState, city.longCountry].join(', '),
+        }),
+        )
+        console.log(geometry)
```
Comment (tyler-whitman) [inline components/InfoBox/CityDetailsInfoBox.js]: Do you want to keep these logs in here? Also one on line 28

**Review by tyler-whitman — APPROVED** [review]
Nothing stands out, code looks good and working as expected.

File: components/InfoBox/CityDetailsInfoBox.js (hunk 1 above)
Comment (danielcolinjames) [reply components/InfoBox/CityDetailsInfoBox.js]: good catch! definitely just a debug log I forgot to delete haha

**Pull request merged**